Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -603,6 +603,44 @@ extension PurchasesOrchestrator: StoreKit1WrapperDelegate {
 
 }
 
+extension PurchasesOrchestrator: PaymentQueueWrapperDelegate {
+
+    func paymentQueueWrapper(
+        _ wrapper: PaymentQueueWrapper,
+        shouldAddStorePayment payment: SKPayment,
+        for product: SK1Product
+    ) -> Bool {
+        // `PurchasesOrchestrator` becomes `PaymentQueueWrapperDelegate` only
+        // when `StoreKit1Wrapper` is not initialized, which means that promoted purchases
+        // need to be handled as a SK2 purchase.
+        // This method converts the `SKPayment` into an SK2 purchase by fetching the product again.
+        assert(self.storeKit1Wrapper == nil, "This method should not be invoked if SK1 is enabled")
```

Comment:
This is not a `precondition` so it will be ignored in release builds. Still useful to catch potential misconfigurations in tests.

Reply:
Should we also log an error if this gets called for any reason with SK1 in release builds?